hphp/runtime/vm/jit/irlower-conv.cpp

```diff
@@ -198,17 +198,23 @@ void cgConvDblToInt(IRLS& env, const IRInstruction* inst) {
   v << cvttsd2siq{src, d};
   v << cmpq{indef, d, sf};
 
+  // For ARM, use the cc which ignores unordered results for the first
+  // condition. Then use the ARM equivalent cc to check for the unordered
+  // result in the second condition.
+  ConditionCode cc_nb = (arch() == Arch::ARM) ? CC_GE : CC_NB;
+  ConditionCode cc_p = (arch() == Arch::ARM) ? CC_O : CC_P;
```

Comment:
Similar to my comment above, if we need a more portable way to represent these special condition codes for ARM, I can add a mechanism to look these up in the abi-xx files for each platform. I didn't in this patch, because I *feel* that the reduced complication of this new patch made these uses of `arch() ==` much more palatable than the previous patch. As I said earlier though, I'm happy to do it. Just let me know.